### PR TITLE
Fix for authentication Xss and permission access issue 

### PR DESCRIPTION
**Pull request by neeraj890kumar**
Resolves: #3553

### CODE REVIEW

File: src-gui/ui/src/app/modules/switches/rule-details/rule-details.component.ts [hunk 1]
```diff
@@ -25,8 +26,14 @@ export class RuleDetailsComponent implements OnInit {
     private toastr: ToastrService,
     private loaderService: LoaderService,
     private clipboardService: ClipboardService,
+    private router:Router,
     public commonService: CommonService
-  ) {}
+  ) {
+    if(!this.commonService.hasPermission('menu_switches')){
+      this.toastr.error('You are not authorised to access this page.');
```
Comment (gauravchug) [inline src-gui/ui/src/app/modules/switches/rule-details/rule-details.component.ts:33]: Remove such messages from code.
Add these error messages in the properties file.

Comment (neeraj890kumar) [reply src-gui/ui/src/app/modules/switches/rule-details/rule-details.component.ts]: added common message constant file

File: src-gui/ui/src/app/modules/switches/switch-detail/switch-detail.component.ts [hunk 2]
```diff
@@ -399,14 +400,16 @@ export class SwitchDetailComponent implements OnInit, AfterViewInit,OnDestroy {
     modalRef.componentInstance.emitService.subscribe(
       evacuate => {
         var data = {"under_maintenance":e.target.checked,"evacuate":evacuate};
-
+          this.loaderService.show('Applying Changes..');
```
[the remark this answers is not in the document]
Comment (neeraj890kumar) [reply src-gui/ui/src/app/modules/switches/switch-detail/switch-detail.component.ts]: added common messages constant file

**Comment by neeraj890kumar** [comment]
> @neeraj890kumar Define messages properties or constant file for error/info messages for UI. Do not hardcode in code.

@gauravchug added messages constant file in gui, hard coded messages are moved to constant file

**Pull request merged**